src/components/TodoItem.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  useContext,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoContext } from './TodoContext';
+
+type Props = { todo: Todo };
+
+export const TodoItem:React.FC<Props> = ({ todo }) => {
+  const { updateTodo, deleteTodo } = useContext(TodoContext);
+  const [switchEditTodo, setSwitchEditTodo] = useState(false);
+  const [newTitle, setNewTitle] = useState('' || todo.title);
+  const todoItemEditRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (switchEditTodo) {
+      todoItemEditRef.current?.focus();
+    }
+  }, [switchEditTodo]);
+
+  const handleUpdateTitle = (tod: Todo) => {
+    if (!newTitle.trim()) {
+      deleteTodo(todo.id);
+      setSwitchEditTodo(false);
+    } else {
+      updateTodo({ ...tod, title: newTitle.trim() });
+      setSwitchEditTodo(false);
+    }
+  };
+
+  const handlesForUpdateTitle = () => {
+    setNewTitle(newTitle);
+    handleUpdateTitle(todo);
+  };
+
+  const handlesCurrentTitle = () => {
+    setNewTitle(todo.title);
+    setSwitchEditTodo(false);
+  };
+
+  const handleOnBlur = () => {
+    if (newTitle === todo.title) {
+      handlesCurrentTitle();
+    } else {
+      handlesForUpdateTitle();
+    }
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      if (newTitle === todo.title) {
+        handlesCurrentTitle();
+      } else {
+        handlesForUpdateTitle();
+      }
+    } else if (e.key === 'Escape') {
+      handlesCurrentTitle();
+    }
```

Comment:
try to simplify your `if` blocks. i didn't tested it, but it should work
```suggestion
    if (e.key !== 'Enter' && e.key !== 'Escape') return;
    
    if (newTitle === todo.title || e.key === 'Escape') {
        handlesCurrentTitle();
    } else {
        handlesForUpdateTitle();
    }
```

Reply:
also pay attention to naming. `handlesCurrentTitle` and `handlesForUpdateTitle` is very bad names because I don't understand what they doing